### PR TITLE
Expand `"type"` to include generic type arguments / parameters

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
- fixes https://github.com/cursorless-dev/cursorless/issues/1231
## Checklist

- [x] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [-] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [-] I have not broken the cheatsheet

### CODE REVIEW

**Review by pokey — DISMISSED** [review]

**Comment by pokey** [comment]
please do decl side before mergning tho

**Comment by pokey** [comment]
what happens with

`foo<Bar<Baz, Bongo>>()`
`Foo<Bar>`
`Bar<Baz, Bongo>`

I think probably ok to support more of these cases

**Comment by pokey** [comment]
It would also be good to have a test to formalize what happens with multiple type args

**Comment by AndreasArvidsson** [comment]
@pokey Ready for second look

File: queries/typescript.core.scm [hunk 1]
```diff
@@ -173,6 +173,32 @@
...
 
+;;!! useState<string>()
+;;!           ^^^^^^
+;;!! useState<Record<string, string>>()
+;;!           ^^^^^^^^^^^^^^^^^^^^^^
+(call_expression
+  type_arguments: (type_arguments
+    (_) @type
+  )
+)
```
Comment (pokey) [inline queries/typescript.core.scm:184]: replace with

```ts
(type_arguments
    (_) @type
  )
```

and params

**Comment by pokey** [comment]
- [ ] add test for `const foo: Record<string, string> = {} `on hte first `string`

**Comment by AndreasArvidsson** [comment]
@pokey updated

**Review by pokey — APPROVED** [review]
🎲🤞

**Pull request merged**